app/model/des/FinancialData.scala

```diff
@@ -25,31 +25,23 @@ import play.api.libs.json.{Json, OFormat, Reads}
  * containing financial data for a VRN.
  */
 
-final case class FinancialData(idType: Option[String], idNumber: Option[String], regimeType: Option[String], processingDate: Option[String], financialTransactions: Option[Seq[Transaction]])
+final case class FinancialData(idType: String, idNumber: String, regimeType: String, processingDate: String, financialTransactions: Option[Seq[Transaction]])
 
 object FinancialData {
   implicit val reads: Reads[FinancialData] = Json.reads[FinancialData]
   implicit val format: OFormat[FinancialData] = Json.format[FinancialData]
 }
 
-final case class Transaction(periodKey:            Option[String],
-                             periodKeyDescription: Option[String],
-                             taxPeriodFrom:        Option[LocalDate],
-                             taxPeriodTo:          Option[LocalDate],
-                             originalAmount:       Option[BigDecimal],
-                             outstandingAmount:    Option[BigDecimal],
-                             items:                Option[Seq[SubItem]])
+final case class Transaction(periodKey:            String,
```

Comment:
It's good practice to encode these domain objects using dedicated types, not strings. Please see my comment related to Vrn.